faqs/_posts/2014-09-23-encrypted-forms.md

```diff
@@ -22,50 +22,107 @@ In order to encrypt your form, you need to generate public and private encryptio
+<br>
+![](/content/screenshots/faq/encrypted-forms-mac-passphrase.png)
+<br>
+4. Then, in order to extract your public key, you need to export it to a MyPublicKey.pem file by typing the following command on the terminal, then pressing Enter:  `openssl rsa -in MyPrivateKey.pem -outform PEM -pubout -out MyPublicKey.pem`
+<br>
+![](/content/screenshots/faq/encrypted-forms-mac-public-key.png)
+<br>
+
+#### On Linux
+
+1. Open terminal by typing Ctrl+Alt+T;
+2. Generate a 2048-bit private key and write it to the MyPrivateKey.pem file by typing the following command on the OpenSSL terminal, then pressing **Enter**: `openssl genpkey -out MyPrivateKey.pem -outform PEM -algorithm RSA -pkeyopt rsa_keygen_bits:2048`
+<br>
+![](/content/screenshots/faq/encrypted-forms-linux-private-key.png)
+<br>
+3. Then, in order to extract your public key, you need to export it to a MyPublicKey.pem file, by typing the following command on the terminal, then pressing **Enter**: `openssl rsa -in MyPrivateKey.pem -inform PEM -out MyPublicKey.pem -outform PEM -pubout`
+<br>
+![](/content/screenshots/faq/encrypted-forms-linux-public-key.png)
+<br>
+
+
+Finally, you have your two keys:
+<br>
+1. **MyPrivateKey.pem** - your private key that you need to save in a secure location. This is the key that you will use to decrypt your data. 
+2. **MyPublicKey.pem** - your public key that you can share with anyone you want to be able to share information securely with. This is the public key that you will use to encrypt your data. Paste this into the public_key field in the XLSForm settings worksheet. 
 
 ### XLSForm syntax for encrypted forms
 
 Follow these steps to add the correct encryption syntax in your XLSForm:
 
 1. Add a worksheet called **settings**;
-1. In this worksheet, create three columns: **id_string**, **submission_url**, and **public_key**. The **id_string** column specifies the form id name.  The **submission_url** is the url that will accept the encrypted form submissions.  This is the same url that you type into the ODK Collect application under settings to download the forms from your account; however, you must add **/submission** to the end of this url.  Finally, the **public_key** column contains the public key (without any spaces or line breaks) that was generated using the OpenSSL software.
+1. In this worksheet, create three columns: **id_string**, **submission_url**, and **public_key**. The **id_string** column specifies the form id name.  The **submission_url** is the url that will accept the encrypted form submissions.  This is the same url that you type into the ODK Collect application under settings to download the forms from your account. However, you must add **/YOUR_USERNAME/submission** to the end of this url.  Finally, the **public_key** column contains the public key (without any spaces or line breaks) that was generated using the OpenSSL software.
```

Comment:
The submission URL should be `/submission` without the username to allow authentication of submissions i.e `https://odk.ona.io/submission`. I think that is what we should be recommending.